chain-signatures/node/src/indexer_eth/mod.rs

```diff
@@ -0,0 +1,1514 @@
+        tokio::spawn(async move {
+            tracing::info!("Spawned task to refresh the latest finalized block");
+            Self::refresh_finalized_block(
+                client_clone,
+                finalized_block_send_clone,
+                refresh_interval,
+            )
+            .await;
+        });
+
+        let node_near_account_id_clone = self.node_near_account_id.clone();
+        let backlog_clone = self.backlog.clone();
+        let client_clone = Arc::clone(&client);
+        let optimistic_requests = self.eth.optimistic_requests;
+        let sign_tx_clone = sign_tx.clone();
+        tokio::spawn(async move {
+            Self::send_requests_when_final(
+                client_clone,
+                requests_indexed_recv,
+                finalized_block_recv,
+                sign_tx_clone,
+                app_data_storage.clone(),
+                node_near_account_id_clone,
+                optimistic_requests,
+                backlog_clone,
+            )
+            .await;
+        });
+
+        let blocks_failed_send_clone = blocks_failed_send.clone();
+        let node_near_account_id_clone2 = self.node_near_account_id.clone();
+        let requests_indexed_send_clone = requests_indexed_send.clone();
+        let backlog_clone2 = self.backlog.clone();
+        let sign_tx_clone = sign_tx.clone();
+        let client_clone = Arc::clone(&client);
+        tokio::spawn(async move {
+            Self::retry_failed_blocks(
+                client_clone,
+                blocks_failed_recv,
+                blocks_failed_send_clone,
+                contract_address,
+                node_near_account_id_clone2,
+                requests_indexed_send_clone,
+                total_timeout,
+                backlog_clone2,
+                sign_tx_clone,
+            )
+            .await;
+        });
+
+        let blocks_to_process_send_clone = blocks_to_process_send.clone();
+        let client_clone = Arc::clone(&client);
+        if let Some(last_processed_block) = last_processed_block {
+            match Self::catchup_end_block_number(client_clone).await {
+                Ok(end_block_number) => {
+                    Self::add_catchup_blocks_to_process(
+                        blocks_to_process_send_clone,
+                        last_processed_block,
+                        end_block_number,
+                    )
+                    .await
+                }
+                Err(err) => {
+                    tracing::error!("Failed to get catchup end block number: {err:?}");
+                }
+            }
+        }
+
+        let blocks_to_process_send_clone = blocks_to_process_send.clone();
+        let client_clone = Arc::clone(&client);
+        tokio::spawn(async move {
+            tracing::info!("Spawned task to add new blocks to process");
+            Self::add_new_block_to_process(client_clone, blocks_to_process_send_clone).await
+        });
```

Comment:
we should put the initial log not inside of an async move block, but inside the functions directly so we can have clean spawn calls like so without polluting our local variable names

```suggestion
        tokio::spawn(Self::add_new_block_to_process(Arc::clone(&client), blocks_to_process_send.clone()));
```